Added documentation for interface support in json-api (#14963)
digital-asset/daml: docs/source/json-api/index.rst

``readers`` may be passed as with :ref:`Query <sync-query-req>`.

> ``templateId`` is optional, but you are strongly advised to always pass it explicitly to minimize the data read from the Ledger API to answer the query. It can be either a template id or a interface id

Reviewer: ```suggestion
``templateId`` is optional, but you are strongly advised to always pass it explicitly to minimize the data read from the Ledger API to answer the query. It can be either a template ID or an interface ID.
```

Where:

> - ``templateIds`` -- either an array of contract template identifiers or an array of single interface identifier to search through. Mixed of template IDs and interface IDs or more than one interface ID are not allowed.

Reviewer: ```suggestion
- ``templateIds`` -- either an array of contract template identifiers or an array containing a single interface identifier to search through. Mixing of template ID's and interface ID's, or more than one interface ID is not allowed.
```

    {"templateIds": ["11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:Iou:Iou"]}

Multiple queries may be specified in an array, for overlapping or
different sets of template IDs.::

    [
        {"templateIds": ["11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:Iou:Iou"], "query": {"amount": {"%lte": 50}}},
        {"templateIds": ["11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:OtherIou:OtherIou"], "query": {"amount": {"%gt": 50}}},
        {"templateIds": ["11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:Iou:Iou"]}
    ]

> Interface ID can be provided in ``templateIds`` but only single interface

Reviewer: ```suggestion
Only one interface ID can be provided in ``templateIds``.
```

> ID can be used in all queries::

Reviewer: ```suggestion
An interface ID can be used in all queries::
```

Author: "ID can be used in all queries::" is following the previous line as 1 sentence. 
I will change it to 
```
Interface ID can be provided in ``templateIds`` but only a single interface
ID can be used in all queries::
```

Author: oh sorry I have mis-read. you actually made suggestion on the previous line
I will apply this

    [
        {"templateIds": ["11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:Ifc:Ifc"], "query": {"amount": {"%lte": 50}}},
        {"templateIds": ["11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:Ifc:Ifc"], "query": {"amount": {"%gt": 50}}},
        {"templateIds": ["11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:Ifc:Ifc"]}
    ]

> Mixed of template IDs and interface IDs or More than one interface ID found in queries are not allowed. BadRequest(400) error will be returned.::

Reviewer: ```suggestion
Mixing of template ID's and interface ID's or More than one interface ID across queries is not allowed. BadRequest(400) error will be returned.::
```
